### ISSUE TITLE
The symbol font setting for the theme is invalid

### ISSUE THREAD

**Issue by 93Alliance** [fyne-io/fyne#3739]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The symbol font setting for the theme is invalid,  creating a label using the symbol font by NewLabelWithStyle does not work.

### How to reproduce

NA

### Example code

```go
func (*CustomTheme) Font(s fyne.TextStyle) fyne.Resource {
	if s.Symbol {
		return symbolFont
	}

	return fontResource
}
```

```go
phonetic := widget.NewLabelWithStyle(ph, fyne.TextAlignLeading, fyne.TextStyle{Symbol: true})
```

### Fyne version

2.3.1

### Go compiler version

1.18

### Operating system and version

linux mint 20.3

**Comment by andydotxyz** [comment]
Can you please be more descriptive about "does not work" - perhaps a screenshot or a description of what happened vs what you expected?

**Comment by 93Alliance** [comment]
I want to show some special symbols, like phonetic symbols.
I want the label component to use the symbol font, but it loads the default symbol font.
The following code should have been executed when the configuration took effect, but it wasn't.
```go
func (*CustomTheme) Font(s fyne.TextStyle) fyne.Resource {
	if s.Symbol {
               // Code can never be executed here
		return symbolFont
	}

	return fontResource
}
```

**Comment by andydotxyz** [comment]
Thanks, this is clear now.

**Comment by andydotxyz** [comment]
Fixed, will be released in v2.4.0
